### PR TITLE
Fix ResourceProcessor to handle multiple Resources

### PR DESCRIPTION
**Pull request by gregturn**
* Bump things up to latest Spring Boot stable release to ensure we're using latest version of Spring Data + Spring HATEOAS
* Fix CitizensResourcesProcessor to use `Resources<Resource<Citizen>>` instead of `Resources<Citizen>`.

### CODE REVIEW

**Comment by gregturn** [comment]
Resolves https://github.com/spring-projects/spring-hateoas/issues/570

File: src/test/java/ResourceProcessorTests.java [hunk 1]
```diff
@@ -86,7 +85,7 @@ public void readListTest() throws Exception {
         mockMvc.perform(get(urlTemplate))
                 .andExpect(status().isOk());
 
-        assertFalse(CitizenResourceProcessor.GOT_CALLED);
+        assertTrue(CitizenResourceProcessor.GOT_CALLED);
```
Comment (FabianWilms) [inline src/test/java/ResourceProcessorTests.java:88]: @gregturn Is this the correct behaviour? That both ResourceProcessors get called? When I'm trying to manipulate some of the entities that get sent out wouldn't this lead to duplicate execution of the code?

**Review by FabianWilms — APPROVED** [review]
So in short to fix the problem: update the spring boot version so that the ResourceProcessor<> gets called for every entity in a list that gets sent out.

Thanks very much for the explanation!

File: src/test/java/ResourceProcessorTests.java (hunk 1 above)
Comment (gregturn) [reply src/test/java/ResourceProcessorTests.java:88]: Yes. 

* `Resources` -based processor is tapped to let you set links on a collection resource. 
* `Resource` -based processor is tapped to let you set links on each individual item's resource.

If there is duplicate execution of code, you must examine what is happening and figure out why.

**Pull request merged**